alica_engine/include/engine/RunnableObject.h

```diff
@@ -81,27 +82,27 @@ class RunnableObject
     virtual void doRun() = 0;
     virtual void doTerminate() = 0;
 
-    void setTracing(TracingType type, std::function<std::optional<std::string>()> customTraceContextGetter = {})
-    {
-        _runnableObjectTracer.setTracing(type, customTraceContextGetter);
-    }
+    void setTracing(TracingType type, std::function<std::optional<std::string>()> customTraceContextGetter = {});
     const std::string& getName() { return _name; };
     IAlicaTrace* getTrace() const { return _runnableObjectTracer.getTrace(); };
 
     void sendLogMessage(int level, const std::string& message) const;
     RunningPlan* getPlanContext() const { return _runningplanContext; }
     const std::shared_ptr<Blackboard> getBlackboard() { return _blackboard; }
     IAlicaWorldModel* getWorldModel() { return _wm; };
+    IAlicaLogger& getLogger() { return _logger; };
```

Comment:
not sure if this is required. The application need not use alica's logger through this interface. It can anyways specify its own logger at init time to the engine